core/src/test/java/org/infinispan/health/ClusterHealthImplTest.java

```diff
@@ -1,107 +1,181 @@
+import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 @Test(testName = "health.ClusterHealthImplTest", groups = "functional")
 public class ClusterHealthImplTest {
 
-    private EmbeddedCacheManager cacheManager;
-    private ClusterHealth clusterHealth;
-
-    @BeforeClass
-    private void init() {
-        GlobalConfigurationBuilder globalConfigurationBuilder = new GlobalConfigurationBuilder().clusteredDefault();
-        globalConfigurationBuilder.transport().clusterName("test").nodeName("test");
-
-        cacheManager = new DefaultCacheManager(globalConfigurationBuilder.build());
-        cacheManager.defineConfiguration("test", new ConfigurationBuilder().build());
-        clusterHealth = new ClusterHealthImpl(cacheManager);
-    }
-
-    @AfterClass
-    private void cleanUp() {
-        if (cacheManager != null) {
-            cacheManager.stop();
-            cacheManager = null;
-        }
-    }
-
-    @Test
-    public void testReturningClusterName() throws Exception {
-        //when
-        String clusterName = clusterHealth.getClusterName();
-
-        //then
-        assertEquals(clusterName, "test");
-    }
-
-    @Test
-    public void testReturningHealthyStatus() throws Exception {
-        //given
-        cacheManager.getCache("test", true);
-
-        //when
-        HealthStatus healthStatus = clusterHealth.getHealthStatus();
-
-        //then
-        assertEquals(healthStatus, HealthStatus.HEALTHY);
-    }
-
-    @Test
-    public void testReturningNodeName() throws Exception {
-        //when
-        String nodeName = clusterHealth.getNodeNames().get(0);
-
-        //then
-        assertTrue(nodeName.contains("test"));
-    }
-
-    @Test
-    public void testReturningNumberOfNodes() throws Exception {
-        //when
-        int numberOfNodes = clusterHealth.getNumberOfNodes();
-
-        //then
-        assertEquals(numberOfNodes, 1);
-    }
-
-    @Test
-    public void testReturningNumberOfNodesWithNullTransport() throws Exception {
-        //given
-        DefaultCacheManager cacheManagerWithNullTransport = mock(DefaultCacheManager.class);
-        clusterHealth = new ClusterHealthImpl(cacheManagerWithNullTransport);
-
-        //when
-        int numberOfNodes = clusterHealth.getNumberOfNodes();
-
-        //then
-        assertEquals(numberOfNodes, 1);
-    }
-
-    @Test
-    public void testReturningNodeNamesWithNullTransport() throws Exception {
-        //given
-        DefaultCacheManager cacheManagerWithNullTransport = mock(DefaultCacheManager.class);
-        clusterHealth = new ClusterHealthImpl(cacheManagerWithNullTransport);
-
-        //when
-        List<String> nodeNames = clusterHealth.getNodeNames();
-
-        //then
-        assertEquals(nodeNames.size(), 0);
-    }
+   private static final String INTERNAL_CACHE_NAME = "internal_cache";
```

Comment:
In very much against what happened here...

We introduced a change in the Health Check logic and in the same Pull Request we rewrote this whole test suite. Some tests have been preserved (though with changed name) and some were wiped out (like `testReturningNodeNamesWithNullTransport` which has been discovered long after implementing this functionality and it appears when using CLI on the server with local caches). In my opinion this is one of the easiest ways to introduce new regressions, which won't be found by the CI because of changed test suite.

Since we extended the health check functionality, I would prefer to add some new tests to this test suite (instead of rewriting it). This way we would be sure that we didn't add any new regressions and at the same time we would test new functionality. If for some reason this test suite can be optimized, please do so, but in a separate Pull Request so that we are 100% we didn't break anything.

Reply:
I payed much attention not removing any test, I just changed to AssertJUnit and renamed.
The original test was mixing thins in a way where tests where getting overlapped with the mocked cache manager. The healty status was testing nothing neither, because even when I deleted lines on the code and test was failing

Reply:
In order to add the fucntionality, I needed to rework on the test

Reply:
moreover, unhealthy or rebalancing status was not tested at all.

Reply:
I first covered the code and after on TDD style, I added tests for internal caches that proved that my new code was failing til I added more core to correct it. Caches where getting created calling this method even if we did not want to create any cache